[GlobalISel] Make IRTranslator able to handle PHIs with empty types.

dfszabo opened the pull request:
SelectionDAG already handle this since e53b7d1a11d180ed7b33190a837d8898ab2a0b71.

arsenm commented on llvm/test/CodeGen/Generic/zero-sized-array.ll:
```diff
@@ -1,4 +1,5 @@
 ; RUN: llc < %s
+; RUN: llc -O0 -mtriple=aarch64 -global-isel < %s
 ; PR9900
 
```
There is not really such a thing as a generic codegen test. You have to have a target, and this demonstrates yet another example of why. This will fail if aarch64 isn't built. You should add a copy in test/AArch64/GlobalISel and not add a target run line here

dfszabo replied:
Okay. Moved and minimized it into `test/AArch64/GlobalISel` folder.

arsenm commented on llvm/test/CodeGen/AArch64/GlobalISel/phi-with-empty-type.ll:
```diff
@@ -0,0 +1,16 @@
+; RUN: llc < %s -O0 -global-isel -mtriple=aarch64
```
Should include checks (probably MIR checks, with -stop-after=irtranslator)

dfszabo replied:
Good idea. Included it.

dfszabo committed:
[GlobalISel] Make IRTranslator able to handle PHIs with empty types.

SelectionDAG already handle this since e53b7d1a11d180ed7b33190a837d8898ab2a0b71.

qcolombet reviewed (approved):
Nice catch!
